src/Plugin.php

```diff
@@ -93,6 +98,18 @@ public static function wp_get_nav_menu_items($items, $menu, $args) {
   }
 
   /**
+   * Removes <a> HTML tag if item link status is checked.
+   *
+   * @implements walker_nav_menu_start_el
+   */
+  public static function walker_nav_menu_start_el($item_output, $item, $depth, $args) {
+    if ((bool) get_post_meta($item->ID, '_menu_item_link_status', TRUE) === FALSE) {
+      $item_output = preg_replace('/<a href=\"(.*?)\">(.*?)<\/a>/', "\\2", $item_output);
```

Comment:
There might be additional markup for links in the generated output as well as link attributes (e.g. for mega menus or dropdown menus), which would be removed by this regex.

Given that it is pretty trivial to regenerate the link item markup (see `/wp-includes/nav-menu-template.php#L178`), I'd rather product the output from scratch instead, replacing the anchor with a span element, so as to retain the attributes on it:

```php
		$item_output = $args->before;
		$item_output .= '<span'. $attributes .'>';
		$item_output .= $args->link_before . $title . $args->link_after;
		$item_output .= '</span>';
		$item_output .= $args->after;
```